lib/iris/tests/__init__.py

```diff
@@ -588,31 +588,18 @@ def assertNoWarningsRegexp(self, expected_regexp=""):
         self.assertFalse(matches, msg)
 
     def _assertMaskedArray(self, assertion, a, b, strict, **kwargs):
-        # Define helper function to extract unmasked values as a 1d
-        # array.
-        def unmasked_data_as_1d_array(array):
-            array = ma.asarray(array)
-            if array.ndim == 0:
-                if array.mask:
-                    data = np.array([])
-                else:
-                    data = np.array([array.data])
-            else:
-                data = array.data[~ma.getmaskarray(array)]
-            return data
-
-        # Compare masks. This will also check that the array shapes
-        # match, which is not tested when comparing unmasked values if
-        # strict is False.
```

Comment:
I'm pretty sure the various array equality functions changed at some point to explicitly check the shape.  I don't remember when though.